src/viadot/sources/onestream.py

```diff
@@ -0,0 +1,485 @@
+"""OneStream API connector for data extraction and management.
+
+This module provides the OneStream class for interacting with OneStream API endpoints,
+including Data Adapter queries, SQL queries, and Data Management sequence execution.
+"""
+
+from itertools import product
+import json
+from typing import Any, Literal
+
+import pandas as pd
+from pydantic import BaseModel
+import requests
+
+from viadot.config import get_source_credentials
+from viadot.sources.base import Source
+from viadot.utils import add_viadot_metadata_columns
+
+
+class OneStreamCredentials(BaseModel):
+    """Check OneStream credentials dict.
+
+    OneStream API Key token is required:
+        - api_token: The unique authentication api token for the OneStream API.
+
+    Args:
+         BaseModel (pydantic.BaseModel): Base class for data validation.
+    """
+
+    api_token: str
+
+
+class OneStream(Source):
+    def __init__(
+        self,
+        server_url: str,
+        application: str,
+        config_key: str | None = None,
+        credentials: OneStreamCredentials | None = None,
+        api_params: dict[str, str] | None = None,
```

Comment:
Don't we already know this is an API source?

```suggestion
        params: dict[str, str] | None = None,
```